plugins/inputs/elasticsearch_query/elasticsearch_query.go

```diff
@@ -0,0 +1,310 @@
+	return `Derive metrics from aggregating Elasticsearch query results`
+}
+
+// Init the plugin.
+func (e *ElasticsearchQuery) Init() error {
+	if e.URLs == nil {
+		return fmt.Errorf("elasticsearch urls is not defined")
+	}
+
+	err := e.connectToES()
+	if err != nil {
+		e.Log.Errorf("E! error connecting to elasticsearch: %s", err)
+		return nil
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(e.Timeout))
+	defer cancel()
+
+	for i, agg := range e.Aggregations {
+		if agg.MeasurementName == "" {
+			return fmt.Errorf("field 'measurement_name' is not set")
+		}
+		if agg.DateField == "" {
+			return fmt.Errorf("field 'date_field' is not set")
+		}
+		e.initAggregation(ctx, agg, i)
+	}
+	return nil
+}
+
+func (e *ElasticsearchQuery) initAggregation(ctx context.Context, agg esAggregation, i int) {
+	// retrieve field mapping and build queries only once
+	mapMetricFields, err := e.getMetricFields(ctx, agg)
+	if err != nil {
+		e.Log.Errorf("not possible to retrieve fields: %v", err.Error())
+		return
+	}
+
+	for _, metricField := range agg.MetricFields {
+		if _, ok := mapMetricFields[metricField]; !ok {
+			e.Log.Infof("metric field '%s' not found on index '%s'", metricField, agg.Index)
+			return
+		}
+	}
+
+	aggregationQueryList, err := e.buildAggregationQuery(mapMetricFields, agg)
+	if err != nil {
+		e.Log.Error(err.Error())
+		return
+	}
+	agg.mapMetricFields = mapMetricFields
+	agg.aggregationQueryList = aggregationQueryList
+	e.Aggregations[i] = agg
+}
+
+func (e *ElasticsearchQuery) connectToES() error {
+	var clientOptions []elastic5.ClientOptionFunc
+
+	if e.esClient != nil {
+		if e.esClient.IsRunning() {
+			return nil
+		}
+	}
+
+	if e.httpclient == nil {
+		httpclient, err := e.createHTTPClient()
+		if err != nil {
+			return err
+		}
+		e.httpclient = httpclient
+	}
+
+	clientOptions = append(clientOptions,
+		elastic5.SetHttpClient(e.httpclient),
+		elastic5.SetSniff(e.EnableSniffer),
+		elastic5.SetURL(e.URLs...),
+		elastic5.SetHealthcheckInterval(time.Duration(e.HealthCheckInterval)),
+	)
+
+	if e.Username != "" {
+		clientOptions = append(clientOptions, elastic5.SetBasicAuth(e.Username, e.Password))
+	}
+
+	if time.Duration(e.HealthCheckInterval) == 0 {
+		clientOptions = append(clientOptions, elastic5.SetHealthcheck(false))
+	}
+
+	client, err := elastic5.NewClient(clientOptions...)
+	if err != nil {
+		return err
+	}
+
+	// check for ES version on first node
+	esVersion, err := client.ElasticsearchVersion(e.URLs[0])
+	if err != nil {
+		return fmt.Errorf("elasticsearch version check failed: %s", err)
+	}
+
+	esVersionSplit := strings.Split(esVersion, ".")
+
+	// quit if ES version is not supported
+	if len(esVersionSplit) == 0 {
+		return fmt.Errorf("elasticsearch version check failed")
+	}
+
+	i, err := strconv.Atoi(esVersionSplit[0])
+	if err != nil || i < 5 {
+		return fmt.Errorf("elasticsearch version %s not supported (currently supported versions are 5.x and 6.x)", esVersion)
```

Comment:
Should this error on 7.x as well?

Reply:
Yes, definitely, current code does not work on 7.x versions